File: t-tasks/src/main/java/com/teo/ttasks/injection/module/activity/MainActivityModule.java
package com.teo.ttasks.injection.module.activity;

import com.teo.ttasks.api.PeopleApi;
import com.teo.ttasks.data.local.PrefHelper;
import com.teo.ttasks.data.remote.TasksHelper;
import com.teo.ttasks.injection.module.fragment.TaskListsFragmentModule;
import com.teo.ttasks.injection.module.fragment.TasksFragmentModule;
import com.teo.ttasks.ui.activities.main.MainActivityPresenter;
import com.teo.ttasks.ui.fragments.task_lists.TaskListsFragment;
import com.teo.ttasks.ui.fragments.tasks.TasksFragment;

import dagger.Module;
import dagger.Provides;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class MainActivityModule {

    @ContributesAndroidInjector(modules = {TasksFragmentModule.class})
    abstract TasksFragment contributeTasksFragmentInjector();

    @ContributesAndroidInjector(modules = {TaskListsFragmentModule.class})
    abstract TaskListsFragment contributeTaskListsFragmentInjector();

    @Provides
    static MainActivityPresenter provideMainActivityPresenter(TasksHelper tasksHelper, PrefHelper prefHelper, PeopleApi peopleApi) {
        return new MainActivityPresenter(tasksHelper, prefHelper, peopleApi);
    }
}

File: t-tasks/src/main/java/com/teo/ttasks/injection/module/activity/MainActivityModule.kt
package com.teo.ttasks.injection.module.activity

import com.teo.ttasks.UserManager
import com.teo.ttasks.api.PeopleApi
import com.teo.ttasks.data.local.PrefHelper
import com.teo.ttasks.data.remote.TasksHelper
import com.teo.ttasks.ui.activities.main.MainActivityPresenter
import dagger.Module
import dagger.Provides

@Module
class MainActivityModule {

    @Provides
    internal fun provideMainActivityPresenter(
        tasksHelper: TasksHelper,
        prefHelper: PrefHelper,
        peopleApi: PeopleApi,
        userManager: UserManager
    ): MainActivityPresenter {
        return MainActivityPresenter(tasksHelper, prefHelper, peopleApi, userManager)
    }
}
